backend/server/acl.go

```diff
@@ -188,6 +188,59 @@ func enforceWorkspaceDeveloperSheetRouteACL(plan api.PlanType, path string, meth
 	return nil
 }
 
+var issueStatusRegex = regexp.MustCompile(`^/issue/(?P<issueID>\d+)/status`)
+
+func enforceWorkspaceDeveloperIssueRouteACL(path string, method string, queryParams url.Values, principalID int, getIssueCreatorIDAndProjectID func(issueID int) (int, int, error), getProjectMemberIDs func(projectID int) ([]int, error)) *echo.HTTPError {
+	if !strings.HasPrefix(path, "/issue") {
+		return nil
+	}
+
+	if method == "GET" {
+		// For /issue route, users other than BytebaseBot cannot list other users' issues.
+		if userStr := queryParams.Get("user"); userStr != "" {
+			if principalID == api.SystemBotID {
+				return nil
+			}
+			userID, err := strconv.Atoi(userStr)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusBadRequest, "Invalid user ID").SetInternal(err)
+			}
+			if principalID != userID {
+				return echo.NewHTTPError(http.StatusUnauthorized, "not allowed to list other users' issues")
+			}
+		}
+	} else if method == "PATCH" {
+		// Workspace developer can only operating the issues if match one of the following conditions:
+		// 1. The creator of the issue.
+		// 2. The member of the project that the issue belongs to.
+		if matches := issueStatusRegex.FindStringSubmatch(path); len(matches) > 0 {
+			issueIDStr := matches[1]
+			issueID, err := strconv.Atoi(issueIDStr)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusBadRequest, "Invalid issue ID").SetInternal(err)
+			}
+			creatorID, projectID, err := getIssueCreatorIDAndProjectID(issueID)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, "Failed to process authorize request.").SetInternal(err)
+			}
+			if creatorID == principalID {
+				return nil
+			}
+			memberIDs, err := getProjectMemberIDs(projectID)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, "Failed to process authorize request.").SetInternal(err)
+			}
+			for _, memberID := range memberIDs {
+				if memberID == principalID {
+					return nil
+				}
+			}
+			return echo.NewHTTPError(http.StatusUnauthorized, "not allowed to operate the issue")
+		}
+	}
+	return nil
```

Comment:
Do we enforce the Get/List/Patch/Create() to require the members of a project?

Reply:
Done.